### ISSUE TITLE
Sort uploaded files properly

### ISSUE THREAD

**Issue by rzach** [Carnap/Carnap#249]
Currently uploaded files are sorted by date they were first uploaded. Could we have Carnap:

- Display the date last updated rather than first uploaded?
- Enable sorting by last saved (up/down)
- Enable sorting by name

**Comment by gleachkr** [comment]
The two new sorting options should now be enabled.

**Comment by rzach** [comment]
Excellent. Can you make the name sort use the same order as the assigned problems (ie "Problem 9" before "Problem 10")?

**Comment by gleachkr** [comment]
Sure thing.

The alternative sort is also imperfect: it's sort-by-length with lexical ordering breaking ties. That does better than staight lexical ordering (for example, it gets "Problem 9" < "Problem 10" right), but it's not quite right, since you could still have something in-between those two not sharing the prefix "Problem". So, there's still room for improvement here.

**Comment by rzach** [comment]
I'm happy! Did you want to leave this open for the date thing?  (I always thought it should give the date of last update not date of first uploaded. But with sort by name I no longer care as much since I now at least have one way to quickly find a file in the dozens I have uploaded)

**Comment by gleachkr** [comment]
I agree about changing the date column to date-of-last-update. So let's leave this open for now.
